content/blog/2025/gcp-filestore-backups/index.md

```diff
@@ -0,0 +1,37 @@
+---
+title: "Announcing backups for GCP-hosted hubs!"
+subtitle: ""
+summary: ""
+authors: ["Sarah Gibson"]
+tags: [open-source]
+categories: [impact]
+date: 2025-02-07T13:08:22+00:00
+lastmod: 2025-02-07T13:10:14+00:00
+featured: false
+draft: false
+
+# Featured image
+# To use, add an image named `featured.jpg/png` to your page's folder.
+# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
+image:
+  caption: ""
+  focal_point: ""
+  preview_only: false
+
+# Projects (optional).
+#   Associate this post with one or more of your projects.
+#   Simply enter your project's folder or file name without extension.
+#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
+#   Otherwise, set `projects = []`.
+projects: []
+---
+
+2i2c are pleased to annouce the development and deployment of automated backups of home directories on GCP-hosted hubs!
+
+We have developed the [`gcp-filestore-backups` project](https://github.com/2i2c-org/gcp-filestore-backups) that regularly creates backups of JupyterHub home directories for disaster recovery purposes. The project is a Python wrapper around the [`gcloud` tool](https://cloud.google.com/sdk/gcloud) to regularly request backups be made of the Filestore hosting JupyterHub's user home directories, by default on a daily basis. The script also manages retention of these backups by checking how recently the last backup was made, and the age of existing backups, by default deleting any backup older than 5 days.
```

Comment:
```suggestion
We have developed the [`gcp-filestore-backups` project](https://github.com/2i2c-org/gcp-filestore-backups) that regularly creates backups of JupyterHub home directories for disaster recovery purposes. The project is a Python wrapper around the [`gcloud` tool](https://cloud.google.com/sdk/gcloud) to regularly request backups be made of the filestore hosting JupyterHub's user home directories, by default on a daily basis. The script also manages retention of these backups by checking how recently the last backup was made, and the age of existing backups, by default deleting any backup older than 5 days.
```

Reply:
Filestore is a noun https://cloud.google.com/filestore?hl=en

Reply:
Ah, but is it a proper noun though? 🤔 

It doesn't even exist as a word in the dictionary (yet!), but to me the capitalisation is a marketing thing for a product name, e.g. Hoover, Kleenex. A Google Filestore is one of those such things, but I think the common usage term filestore is acceptable.

Feel free to accept or reject as you wish -- this is more of a style suggestion and you have the final say as the author!

Reply:
Right, but I don't think a filestore is a thing outside of Google's marketing. We'd just call it storage, or use Amazon's marketing name for example. But since this solution is specific to hubs using Google's Filestore, it seems appropriate. E.g. I'll be developing a whole new back-up solution as part of the per-user storage on GCP initiative, because that won't use Filestore and this project won't work for it.